tapir/coop/tests/membership_resignation/test_service.py

```diff
@@ -17,5 +34,124 @@ def setUp(self) -> None:
         self.given_feature_flag_value(feature_flag_membership_resignation, True)
         mock_timezone_now(self, self.NOW)
 
-    def test_todo(self):
-        self.fail("TODO, test service functions.")
+    def test_update_shifts_and_shares_and_pay_out_day_SharesAndSettingPayOutDay_transferredAndSet(self):
+        resignations = [
+            MembershipResignation.ResignationType.BUY_BACK,
+            MembershipResignation.ResignationType.GIFT_TO_COOP,
+            MembershipResignation.ResignationType.TRANSFER,
+        ]
+
+        self.login_as_member_office_user()
+        for resignation in resignations:
+            share_owner = ShareOwnerFactory.create(nb_shares=2)
+            resignation = MembershipResignationFactory.create(
+                share_owner=share_owner,
+                resignation_type=resignation,
+            )
+            MembershipResignationService.update_shifts_and_shares_and_pay_out_day(
+                resignation=resignation,
+            )
+            shares_after_update = resignation.share_owner.share_ownerships
+            match resignation:
+                case MembershipResignation.ResignationType.BUY_BACK:
+                    for share in shares_after_update.all():
+                        self.assertEqual(
+                            resignation.cancellation_date.replace(
+                                day=31, month=12, year=resignation.cancellation_date.year + 3
+                            ),
+                            share.end_date,
+                        )
+                case MembershipResignation.ResignationType.GIFT_TO_COOP:
+                    for share in shares_after_update.all():
+                        self.assertEqual(share.end_date, resignation.cancellation_date)
+                case MembershipResignation.ResignationType.TRANSFER:
+                    self.assertEqual(ShareOwnership.objects.get(
+                        share_owner=resignation.transferring_shares_to,
+                        start_date=resignation.cancellation_date).count(),
+                        shares_after_update.count()
+                        )
+                    self.assertEqual(share.end_date, resignation.cancellation_date)
+
+    def test_update_shifts_shiftsAndShiftAttendance_cancelled(self):
```

Comment:
We need one more test on `update_shifts` to make sure that shifts that are before the end of the membership don't get cancelled.